src/API/Router.ts

```diff
@@ -39,8 +41,16 @@ export class Router {
     this.configuration = configuration
     this.workController = workController
 
-    this.koaRouter.get('/works/:id', this.getWork)
-    this.koaRouter.get('/works', this.getWorks)
+    const getWorksSchema = {
```

Comment:
Just thought: what if we move the `params` and `query` that we're defining inline in the call to `RequestValidationMiddleware` to here?

So it would look something like this:

```ts
const getWorksSchema = {
  params: {
    id: Joi.string().required(),
  }
}
...
this.koaRouter.get('/works/:id', RequestValidationMiddleware(getWorksSchema), this.getWork)
```

This way, the `getWorksSchema` is actually the complete schema for that endpoint. Otherwise, we're really defining part of the schema here only, and the last bit inline in the call to `RequestValidationMiddleware`.

Well, that's my reasoning, anyway. 

Let me know what you think.

Reply:
yes, you can define both ways. I think it's more simple like as you say.